### PR TITLE
Properly build GHC 7.x and some packages in the Darwin sandbox

### PR DESCRIPTION
**Pull request by pikajude**
For GHC 6.x, `cannot bootstrap GHC on this platform` so I left it alone.

### CODE REVIEW

File: pkgs/development/haskell-modules/configuration-common.nix [hunk 1]
```diff
@@ -127,6 +128,21 @@ self: super: {
   # https://github.com/haskell/vector/issues/47
   vector = if pkgs.stdenv.isi686 then appendConfigureFlag super.vector "--ghc-options=-msse2" else super.vector;
 
+  # cabal2nix likes to generate dependencies on hinotify when hfsevents is really required
+  # on darwin
+  hinotify = if pkgs.stdenv.isDarwin then super.hfsevents else super.hinotify;
+
+  # Doesn't properly handle nonsense byte sequences on HFS+
+  system-fileio = dontCheck super.system-fileio;
```
Comment (bennofs) [inline pkgs/development/haskell-modules/configuration-common.nix]: If it's only on HFS+, maybe disable tests conditionally only on darwin? (Or does any other platform have HFS+)? Same for fsnotify tests

**Comment by joelteon** [comment]
Whoops, I neglected to apply the fixes to all the GHCs, pushing some commits to fix

**Comment by copumpkin** [comment]
:+1: from me, as usual!

**Comment by copumpkin** [comment]
Paging @peti, @jwiegley, @shlevy, and other parties interested in both (or either) Haskell and Darwin :smile:

File: pkgs/development/haskell-modules/configuration-common.nix (hunk 1 above)
Comment (peti) [inline pkgs/development/haskell-modules/configuration-common.nix]: Could you please report these issues to the library authors (and possibly even add links to the tickets here in a comment)?

**Comment by peti** [comment]
Test builds are running at http://hydra.cryp.to/jobset/nixpkgs/haskell-updates.

**Comment by copumpkin** [comment]
@peti that confuses me. It looks like the number of jobs decreased substantially but with no failures?

**Comment by bennofs** [comment]
@copumpkin Peti's haskellPackages.nix has an option `big` that enables / disables some of the packages which are build. What you're seeing is with `big = false` (it was probably true before). Also this is build on x86_64-linux, btw.

**Comment by peti** [comment]
I condensed commits 79d0a6f, 771438e, 4933994, afb33e5k into one and pushed them to `staging` as db4cb021c024b65c08a3791edf2d4c613e8d44d8.

I didn't merge commits 37b64b7 and 78bd05e yet, because there've been some comments about them in the reviews that I feel should be addressed first.

@copumpkin, as @bennofs said, I disabled most of the builds because rebuilding all of Hackage takes a long time. Instead, I just built important packages like `git-annex`and `pandoc` with the new compilers. The test build included some packages for *all* active compilers, though, not just ghc 7.8.4. All I can say is that these patches have no negative affect for Linux, though, I can't run Darwin builds on my Hydra instance.

**Comment by copumpkin** [comment]
@peti that makes sense, thanks! This will significantly shrink pure-darwin's remaining diff :grin:

**Comment by peti** [comment]
@joelteon, do you intend to address those review comments for the remaining changes?

**Comment by joelteon** [comment]
I added comments to explain the overrides. What else is needed?

**Comment by peti** [comment]
When a test suite fails and needs to be disabled, please report the issue upstream and add a link pointing to the appropriate ticket to the override, if possible. Also, if a test suite fails only on Darwin, please disable it for Darwin only.

**Comment by joelteon** [comment]
@peti opened issues and added comments

File: pkgs/development/compilers/ghc/7.0.4.nix [hunk 2]
```diff
@@ -13,6 +13,8 @@ stdenv.mkDerivation rec {
 
   buildInputs = [ ghc perl gmp ncurses ];
 
+  enableParallelBuilding = true;
```
Comment (peti) [inline pkgs/development/compilers/ghc/7.0.4.nix]: I am worried that enabling parallelism in older expression will cause trouble. We had parallel builds enabled in the past, and it did us no good: cb21683794269846ba76e66a9c3b1869ad4b817d.

**Comment by peti** [comment]
Merged in 7e7a98c and cd5ef99.

**Pull request closed without merging**